src/shared/TestInfrastructure/Objects/TestGitConfiguration.cs

```diff
@@ -1,22 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
 using System;
+using System.Collections;
 using System.Collections.Generic;
 
 namespace Microsoft.Git.CredentialManager.Tests.Objects
 {
-    public class TestGitConfiguration : IGitConfiguration
+    public class TestGitConfiguration : IGitConfiguration, IDictionary<string, string>
     {
-        public string RepositoryPath { get; set; }
+        private readonly IDictionary<string, string> _dict;
+
+        public TestGitConfiguration() : this(null, null) { }
+
+        public TestGitConfiguration(string repositoryPath) : this(repositoryPath, null) { }
+
+        public TestGitConfiguration(IDictionary<string,string> config) : this(null, config) { }
+
+        public TestGitConfiguration(string repositoryPath, IDictionary<string,string> config)
+        {
+            RepositoryPath = repositoryPath;
+            _dict = config ?? new Dictionary<string, string>();
+        }
 
-        public IDictionary<string, string> Values { get; set; } = new Dictionary<string, string>();
+        public string RepositoryPath { get; set; }
 
         #region IGitConfiguration
 
         string IGitConfiguration.RepositoryPath => RepositoryPath;
 
-        bool IGitConfiguration.TryGetString(string name, out string value) => Values.TryGetValue(name, out value);
+        bool IGitConfiguration.TryGetValue(string name, out string value) => _dict.TryGetValue(name, out value);
 
         void IDisposable.Dispose() { }
 
         #endregion
+
+        #region IDictionary<string, string>
+
+        public IEnumerator<KeyValuePair<string, string>> GetEnumerator() => _dict.GetEnumerator();
```

Comment:
I'm not sure I love needing to duplicate the IDictionary interface. Is there a simpler option here?

Reply:
This is just a testing connivence more than anything, to avoid having the `.Dictionary` everywhere.
I could just keep the commonly used methods (`TryGetValue`, `this[string key]`) instead and also expose the internal dictionary for more complicated usages.